redeploy-kyrix-server

```diff
@@ -13,27 +15,47 @@ fi
 for elem in $active; do
-    if ((`echo $active | grep -c $elem` > 0 )); then kubectl delete $elem; echo "deleting $elem..."; fi
+    if ((`echo $active | grep -c $elem` > 0 )); then echo "deleting $elem..."; kubectl delete $elem; fi
 done
 while [ 1 ]; do w=`kubectl get -o name pods,deployments,services,pvc | egrep kyrix|wc -l`; if [ $w = 0 ]; then break; fi; spin "waiting for $w kubernetes resources (pods, etc) to shut down"; done
 
 kubectl create -f kyrix-deployment.yaml
 
 while [ 1 ]; do w=`kubectl get pods -o wide | egrep "kyrix.+Running" |wc -l`; if [ $w = 1 ]; then break; fi; spin "waiting for kyrix kubernetes pod to start"; done
 
-source setup-kyrix-vars.env
+source setup-kyrix-vars.env > /dev/null
 
+# expose early because it takes a while to allocate an external IP address
 kubectl expose replicaset.apps/$RS_KYRIX --type=LoadBalancer --name=kyrixserver --port=8000 --target-port=8000
 
-# provide telnet, ping, emacs
-kubectl exec -it $KYRIX -- apk add -y busybox-extras emacs > /dev/null 2>&1
-cat docker-scripts/start-kyrix.sh | kubectl exec -i $KYRIX -- sh -c "cat > /start-kyrix.sh"
+# development
+if [ "x$DEV_NBA" = "x1" ]; then
+    echo "recreating nba db..."
+    recreatedb-kyrix nba
+    export DEV=1
+fi
+if [ "x$DEV_COMPILER" = "x1" ]; then
+    echo "copying tile-server code..."
+    cat docker-scripts/start-kyrix.sh | kubectl exec -i $KYRIX -- sh -c "cat > /start-kyrix.sh; chmod 755 /start-kyrix.sh"
+    tar cf - back-end/src | kubectl exec -it $KYRIX -- sh -c "cd ..; tar xvf -"
+    echo "recompiling tile-server code..."
+    kubectl exec -it $KYRIX -- bash -c 'cd /kyrix/back-end; mvn -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn compile | tee mvn-compile.out >/dev/null 2>&1'
+    export DEV=1
+fi
+if [ "x$DEV" = "x1" ]; then
+    echo "adding dev tools (telnet, ping, emacs, etc) on the kyrix server..."
+    kubectl exec -it $KYRIX -- apk add -y bash busybox-extras emacs > /dev/null 2>&1
+    echo "re-copying local start-kyrix.sh..."
+    kubectl cp docker-scripts/start-kyrix.sh $KYRIX:/
+fi
 
-kubectl exec -it $KYRIX -- sh -c "cd /kyrix/back-end; PGHOST=master POSTGRES_PASSWORD=kyrixftw USERNAME=kyrix USER_PASSWORD=kyrix_password /wait-for-postgres master:5432 -t 60 -- /start-kyrix.sh" &
+DATA=$DATA ./restart-kyrix-server.sh &
 
+# wait for external IP then export vars
 while [ 1 ]; do ip=`kubectl get services -o wide | grep kyrixserver | awk '{print $4}'`; if [ $ip != '<pending>' ]; then break; fi; spin "waiting for external IP"; done
+source setup-kyrix-vars.env > /dev/null
 
-source setup-kyrix-vars.env
+while [ 1 ]; do started=`kubectl exec -it $KYRIX -- sh -c "cat /kyrix-started | tr -d '\n' "`; if [ "x$started" == 'xyes' ]; then break; fi; spin "waiting for kyrix to start"; done
 
 echo "Kyrix running; run 'source setup-kyrix-vars.env' for convenience scripts/functions or visit http://$KYRIX_IP:8000/"
```

Comment:
I found printing `$KYRIX_IP` here pretty useful. However, when I run `restart-kyrix-server.sh` standalone (for dots), it won't tell me the IP. 

I wonder if we can move this while-waiting-for-kyrix-to-start loop to `restart-kyrix-server.sh` without breaking anything.

Reply:
do it in both places?

Reply:
ok I will do it.

Reply:
turns out that having it in both places would make redeploy-kyrix-server print twice. So I just moved it to `restart-kyrix-server.sh` and it worked for both cases.